requirements/dev/build-requirements.txt

```diff
@@ -0,0 +1,6 @@
+pybind11
+meson-python
+cmake
+ninja
```

Comment:
What do we need ninja for?

Reply:
If I remember correctly I got this list in the chat on gitter while triaging the first time. We can remove it if it is unnecessary.

Reply:
Woah! Removing ninja has failed the CI build😮

Reply:
> I would have expected that meson-python declaration ninja as a dependency.

It effectively does, if you are using build isolation, but not if you aren't. See contourpy/contourpy#260 for discussion with a `meson` maintainer and follow-up in contourpy/contourpy#270.

There is no good answer to explicitly including `ninja` in the build requirements file. If you don't include it and are not using build isolation then you may well need to explicitly install it, particularly on macos and windows. But if you do include it there are situations such as on cygwin where there are no wheels available and you will end up trying to build it from source, which may not be a pleasant experience, rather than using the prebuilt cygwin package (e.g. #26543).

I have a slight preference for not including it, so that `build-requirements.txt` is consistent with the `[build-system]` section of `pyproject.toml`. This does mean that some CI runs may require it to be explicitly installed using `pip`, `sudo apt`, or whatever.

Reply:
Do you mean that we need to put ninja dependency manually in the Ci scripts, separete from build-requirements.txt?

Reply:
We might need to, but shouldn't do it by default. We should only consider doing so in specific CI runs that fail at the build stage due to lack of `ninja`.

Reply:
I have removed both ninja and cmake. I see a test coverage pipeline failing. Others have passed.